### ISSUE TITLE
@postConstruct not called in parent class

### ISSUE THREAD

**Issue by sanex3339** [inversify/InversifyJS#615]
`@postConstruct` not called in parent class.

```ts
@injectable()
abstract class Abstract {
     @postConstruct()
     initialize () {//...} // <- not called
}

@injectable()
class Concrete extends Abstract {
    //...
}
```

`target` property in decorator pointed on parent class, not on concrete class which extends this parent class: https://github.com/inversify/InversifyJS/blob/master/src/annotation/post_construct.ts#L12

But in `instantination.ts` in condition - metadata property key is finding in constructor of concrete class:
https://github.com/inversify/InversifyJS/blob/master/src/resolution/instantiation.ts#L37

## Expected Behavior
`@postConstruct` should called in parent class.

## Current Behavior
`@postConstruct` not called in parent class.

## Steps to Reproduce (for bugs)
Factory declaration:
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/container/InversifyContainerFacade.ts#L59

Factory binding:
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/container/modules/node-transformers/NodeTransformersModule.ts#L81

`@postConstruct` decorated method in parent class:
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/a3dad6cf948edead20078e598b5ffcb943a2dc6c/src/node-transformers/AbstractNodeTransformer.ts#L41

## Your Environment

* Version used: latest version
* Environment name and version (e.g. Chrome 39, node.js 5.4): node@8.0.0
* Link to your project: https://github.com/javascript-obfuscator/javascript-obfuscator

**Comment by sanex3339** [comment]
So, i think we should use `hasMetadata` instead of `hasOwnMetadata` here:
https://github.com/inversify/InversifyJS/blob/master/src/resolution/instantiation.ts#L37

**Comment by sanex3339** [comment]
I'll try to make PR

**Comment by remojansen** [comment]
Fixed by #616
